internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -126,7 +127,21 @@ func (ws *workflows) StartWorkflow(ctx context.Context, session sdktypes.Session
 
 func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkflowParams) error {
 	wi := workflow.GetInfo(wctx)
-	sid := params.Data.Session.ID()
+	s := params.Data.Session
+	sid := s.ID()
+
+	// get session time (it was created not long after event dispatch) and then update it with actual event time if found
+	eventTime := s.CreatedAt()
+	if eWr, ok := s.Inputs()["event"]; ok {
+		if eUnWr, err := sdktypes.UnwrapValue(eWr); err == nil {
+			if eMap, ok := eUnWr.(map[string]any); ok {
+				if createdAt, ok := eMap["created_at"].(time.Time); ok {
+					eventTime = createdAt
```

Comment:
why do you need this time if you check the session time ?

Reply:
explain?

Reply:
either user session or event time depends on the requirement

Reply:
we talked. removed session time. only event time will be used